ReleaseNotes.md

```diff
@@ -1,5 +1,24 @@
 # Release Notes
 
+## 2.0.0
+
+- Add support for `net462`, `net7.0` and `net8.0`
+- **[BREAKING CHANGE]** Drop support for `net461`, `netcoreapp3.1` and `net5.0`, all of which are EOL
+- Added `SourceLink` support to improve consumer debugging experience
+- Upgrade nuget packages to latest versions
+- Updated `.gitignore` using `dotnet new gitignore`
+- Added `.editorconfig` using `dotnet new editorconfig`
+- Added `global.json` with SDK version configuration
+- Moved solution file to project root
+- Added recommended VS Code extensions to `extensions.json`
+- Added VS Code settings for formatting and file associations
+- Added tool manifest using `dotnet new tool-manifest`
+- Added `dotnet-format` to tool manifest
+- Added ReSharper/Rider config
+- Formatted `README.md`, `Benchmark.md`, `ReleaseNotes.md`, `Directory.Build.props` files, `csproj` files, and `test/Examples` json files
+- Ran `dotnet format` for error, warn, and info severity
+- Moved test `TargetFrameworks` to `Directory.Build.props` for test folder
```

Comment:
I appreciate all the valuable changes you're proposing, but can we have separate contributions (backed by issues created first please) to address the following:
- adding dotnet-format
- adding support for VS Code contributors

Those non-functional changes wouldn't be necessary listed in the release notes as the release notes are publicly visible to the package users. Please only keep them in the description of the PR.

Reply:
This is completely fair. I've stripped back the changes in this PR to be much more focused on specifically adding [SourceLink](https://learn.microsoft.com/en-us/dotnet/standard/library-guidance/sourcelink) and I will raise a separate PR to add `dotnet-format` along with the default .NET editorconfig.